### PR TITLE
[#27] feat: log in 

### LINKED ISSUE #27: [Backend] Login as a author

### PR DESCRIPTION
**Pull request by IanSeng**
# Summary 
This pr is to create author signing endpoint, get endpoint and minor fixes on signup function. Instead of passing username and password, an authentication token will be returned once user is signin and it can used as credential for making backend request. 

**Documentation**:  [user wikipage](https://github.com/C404TEAMW21/CMPUT404_PROJECT/wiki/User

## Sub-tasks 
- [x] Author profile get endpoint 
- [x] User signing function  
- [x] Return authentication token 
- [x] Unit testing
- [x] Fix sign up return object 
- [x] Update documentation 
- [x] Update readme

## Resolve 
Resolved #27 [Backend] Login as author

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,3 +126,15 @@ Django Settings
 AssertRegex
 * From [StackOverflow](https://stackoverflow.com/a/64192098)
 * Accessed Feb 14 2021
+
+Django Restframework
+* From [Bennett Garner](https://medium.com/swlh/build-your-first-rest-api-with-django-rest-framework-e394e39a482c)
```
Comment (bui1) [inline README.md:131]: Can we seperate this to 
* From [Bennett Garner](their medium profile page)
* From [Medium](link to where you found it)

File: README.md [hunk 2]
```diff
@@ -126,3 +126,15 @@ Django Settings
...
+* Accessed Feb 15 2021
+
+Fix Django Authenticate Token issue 
+* From [ilse2005](https://stackoverflow.com/questions/35601130/django-rest-framework-using-tokenauthentication-with-browsable-api)
```
Comment (bui1) [inline README.md:135]: Similar to comment above, seperate the stack overflow user from the actual page you got this from

File: server/author/tests/test_create_author_endpoint.py [hunk 3]
```diff
@@ -62,6 +70,85 @@ def test_create_user_with_password_too_short(self):
...
-        res = self.client.post(SIGNUP_USER_URL, payload)
+        res = self.client.post(CREATE_USER_URL, payload)
+
+        self.assertEqual(res.status_code, status.HTTP_400_BAD_REQUEST)
+
+class TestAuthAuthorEndpoint(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_author_endpoint_to_return_token(self):
```
Comment (bui1) [inline server/author/tests/test_create_author_endpoint.py:81]: could we also add docstrings for each of these unit tests? It makes it easier to debug in the console later
Something like

def test_auth_author_endpoint_to_return_token(self):
   """Test if author endpoint returns token properly"""

**Review by bui1 — CHANGES_REQUESTED** [review]
Nice work. Just a couple of housecleaning tasks :)

File: server/author/tests/test_create_author_endpoint.py (hunk 3 above)
Comment (IanSeng) [reply server/author/tests/test_create_author_endpoint.py:86]: addressed but GitHub did not pick it up because it was a new line 😭.

**Comment by IanSeng** [comment]
> Nice work. Just a couple of housecleaning tasks :)

Those are very good feedbacks and practices!!! I have addressed and thank you!!! 🎉

**Review by bui1 — APPROVED** [review]
Nice work!

**Pull request merged**